python-sdk/src/astro/sql/operators/raw_sql.py

```diff
@@ -44,9 +58,30 @@ def execute(self, context: Context) -> Any:
                 "backend."
             )
 
-        if self.handler and self.database_impl.sql_type == "delta":
-            return result
+        # if a user specifies a results_format, we will return the results in that format
+        # note that it takes precedence over the handler
+        if self.results_format:
+            conversion_func = getattr(self, f"results_as_{self.results_format}")
+
+            # if fail_on_empty is set to False, wrap in a try/except block
+            # this is because the conversion function will fail if the result is empty
+            # due to sqlalchemy failing when there are no results
+            if not self.fail_on_empty:
+                try:
+                    return conversion_func(result)
+                except Exception:
+                    return None
+
+            # otherwise, just return the result
+            return conversion_func(result)
+
+        # if no results_format is specified, we will return the results in the format specified by the handler
         elif self.handler:
+            # if it's a delta table, we don't need to do any conversion
+            if self.database_impl.sql_type == "delta":
+                return result
```

Comment:
This should go to the DeltaDatabase class. So create a method/property on BaseDatabase called "format_result_handler_conversion" or something like that, which should probably just have `pass` in BaseDatabase. And then you override it in DeltaDatabase.

So then you can do:

```python
self.database_impl.format_result_handler_conversion(result)
```

So that we don't have "tech"-specific loginc in Operators

Reply:
this was originally introduced as part of @dimberman's databricks work, so he agreed to address this comment as a follow on to this PR. filed an issue to track it [here](https://github.com/astronomer/astro-sdk/issues/1606)

Reply:
Sounds good